File: scratch/test_avatar_upload.js
const https = require('https');
const fs = require('fs');
const path = require('path');

const postData = JSON.stringify({
  email: 'student2@edusphere.com',
  password: 'Password@123'
});

const reqOptions = {
  hostname: 'edusphere-erp-frontend.onrender.com',
  path: '/api/v1/auth/login',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': postData.length
  }
};

const req = https.request(reqOptions, (res) => {
  let body = '';
  res.on('data', (d) => { body += d; });
  res.on('end', () => {
    try {
      const loginRes = JSON.parse(body);
      if (!loginRes.success) {
        console.error('Login failed:', loginRes);
        return;
      }
      const token = loginRes.token;
      const userId = loginRes.user.id;
      console.log('Login successful! Token acquired. User ID:', userId);

      // Create a dummy image file
      const dummyFilePath = path.join(__dirname, 'dummy.png');
      fs.writeFileSync(dummyFilePath, 'dummy image binary content');

      // Prepare Multipart payload
      const boundary = '----WebKitFormBoundary7MA4YWxkTrZu0gW';
      const filename = 'dummy.png';
      const fileField = 'avatar';

      let multipartBody = '';
      multipartBody += `--${boundary}\r\n`;
      multipartBody += `Content-Disposition: form-data; name="${fileField}"; filename="${filename}"\r\n`;
      multipartBody += 'Content-Type: image/png\r\n\r\n';
      
      const fileData = fs.readFileSync(dummyFilePath);
      const footer = `\r\n--${boundary}--\r\n`;

      const totalLength = Buffer.byteLength(multipartBody) + fileData.length + Buffer.byteLength(footer);

      const uploadOptions = {
        hostname: 'edusphere-erp-frontend.onrender.com',
        path: `/api/v1/users/${userId}/avatar`,
        method: 'PATCH',
        headers: {
          'Authorization': `Bearer ${token}`,
          'Content-Type': `multipart/form-data; boundary=${boundary}`,
          'Content-Length': totalLength
        }
      };

      const uploadReq = https.request(uploadOptions, (uploadRes) => {
        let uBody = '';
        uploadRes.on('data', (d) => { uBody += d; });
        uploadRes.on('end', () => {
          console.log('Upload status:', uploadRes.statusCode);
          console.log('Upload response:', uBody);
          
          // Clean up dummy file
          try { fs.unlinkSync(dummyFilePath); } catch(_) {}
        });
      });

      uploadReq.on('error', (err) => {
        console.error('Upload request error:', err);
      });

      uploadReq.write(multipartBody);
      uploadReq.write(fileData);
      uploadReq.write(footer);
      uploadReq.end();

    } catch (e) {
      console.error('Failed to parse login JSON:', body);
    }
  });
});

req.on('error', (e) => {
  console.error(e);
});

req.write(postData);
req.end();

File: scratch/test_avatar_upload.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

void main() async {
  final loginUrl = Uri.parse('https://edusphere-erp-frontend.onrender.com/api/v1/auth/login');
  final loginRes = await http.post(
    loginUrl,
    headers: {'Content-Type': 'application/json'},
    body: jsonEncode({
      'email': 'student2@edusphere.com',
      'password': 'Password@123',
    }),
  );
  
  print('Login Response: ${loginRes.statusCode} - ${loginRes.body}');
  final loginData = jsonDecode(loginRes.body);
  if (loginData['success'] != true) {
    return;
  }
  
  final token = loginData['token'];
  final userId = loginData['user']['id'];
  
  final uploadUrl = Uri.parse('https://edusphere-erp-frontend.onrender.com/api/v1/users/$userId/avatar');
  final request = http.MultipartRequest('PATCH', uploadUrl);
  request.headers['Authorization'] = 'Bearer $token';
  
  request.files.add(
    http.MultipartFile.fromBytes(
      'avatar',
      [137, 80, 78, 71, 13, 10, 26, 10, 0, 0, 0, 13, 73, 72, 68, 82, 0, 0, 0, 1, 0, 0, 0, 1, 8, 6, 0, 0, 0, 31, 21, 108, 16, 0, 0, 0, 10, 73, 68, 65, 84, 120, 156, 99, 0, 1, 0, 0, 5, 0, 1, 13, 10, 45, 180, 0, 0, 0, 0, 73, 69, 78, 68, 174, 66, 96, 130],
      filename: 'test.png',
    ),
  );
  
  final response = await request.send();
  final responseBody = await response.stream.bytesToString();
  print('Upload Response Status: ${response.statusCode}');
  print('Upload Response Body: $responseBody');
}
